lib/explorer/data_frame.ex

```diff
@@ -5946,6 +5949,52 @@ defmodule Explorer.DataFrame do
     Shared.apply_impl(df, :covariance, [out_df, opts[:ddof]])
   end
 
+  # SQL
+
+  @doc """
+  Create a dataframe from the result of a SQL query on an existing dataframe.
+
+  > ### SQL Query is Unvalidated {: .warning}
+  >
+  > Explorer does not validate the SQL query string provided. Instead it passes
+  > it directly to the backend. As such, the SQL dialect will be backend
+  > dependent and any errors will come directly from the backend.
+
+  ## `from` Clause
+
+  The `from` clause of the SQL query should reference a chosen table name. The
+  default name is `"df"`. See the examples for a custom table name.
+
+  ## Examples
+
+  Basic example:
+
+      iex> df = Explorer.DataFrame.new(a: [1, 2, 3], b: ["x", "y", "y"])
+      iex> Explorer.DataFrame.sql(df, "select a, b from df group by b order by b")
+      #Explorer.DataFrame<
+        Polars[2 x 2]
+        a list[s64] [[1], [2, 3]]
+        b string ["x", "y"]
+      >
+
+  Custom table name:
+
+      iex> df = Explorer.DataFrame.new(a: [1, 2, 3])
+      iex> Explorer.DataFrame.sql(df, "select a + 1 from my_table", table_name: "my_table")
```

Comment:
I'm curious if it will propagate to the next SQL call involving this resultant DF. Does Polars saves it with the DF?

Reply:
Great question. Thankfully Polars does what you'd hope:

```elixir
iex(1)> df1 = Explorer.DataFrame.new(a: [1, 2, 3])
#Explorer.DataFrame<
  Polars[3 x 1]
  a s64 [1, 2, 3]
>

iex(2)> df2 = Explorer.DataFrame.sql(df1, "select a + 1 from my_table", table_name: "my_table")
#Explorer.DataFrame<
  Polars[3 x 1]
  a s64 [2, 3, 4]
>

iex(3)> df3 = Explorer.DataFrame.sql(df2, "select a + 1 from my_table", table_name: "my_table")
#Explorer.DataFrame<
  Polars[3 x 1]
  a s64 [3, 4, 5]
>
```

You actually can use the `polars::sql::SQLContext` to remember old dataframes, join multiple dataframes, etc. I'm using it very simplistically: creating a new one on every call to `DF.sql`.

Reply:
I see. Nice! We can try to use that more in the future. I mean, save it to the `ExDataFrame` struct and propagate.

Reply:
Yep! They have a concept of "registering" tables, which is how you do multi-table stuff. So we could either wrap that functionality in our own API or just expose the underlying functionality as is.